src/Angor/Client/wwwroot/ServiceWorkerRegister.js

```diff
@@ -0,0 +1,74 @@
+﻿const serviceWorkerFileName = 'service-worker.js';
+const swInstalledEvent = 'installed';
+const staticCachePrefix = 'angor-cache-v';
+const blazorAssembly = 'Angor';
+const blazorInstallMethod = 'PWAInstallable';
+
+const notifyNewVersion = () => {
+    const bc = new BroadcastChannel('angor-channel');
+    bc.postMessage('new-version-found');
+}
+
+window.updateAvailable = new Promise(function (resolve, reject) {
+    if ('serviceWorker' in navigator) {
+        navigator.serviceWorker.register(serviceWorkerFileName)
+            .then(function (registration) {
+                console.log('Registration successful, scope is:', registration.scope);
+                registration.onupdatefound = () => {
+                    const installingWorker = registration.installing;
+                    installingWorker.onstatechange = () => {
+                        switch (installingWorker.state) {
+                            case swInstalledEvent:
+                                if (navigator.serviceWorker.controller) {
+                                    resolve(true);
+                                } else {
+                                    resolve(false);
+                                }
+                                break;
+                            default:
+                        }
+                    };
+                };
+            })
+            .catch(error =>
+                console.log('Service worker registration failed, error:', error));
+    }
+});
+window['updateAvailable']
+    .then(isAvailable => {
+        if (isAvailable) {
+            notifyNewVersion();
+        }
+    });
+
+function showAddToHomeScreen() {
+    setTimeout(function () {
+        const installButton = document.getElementById('installButton');
+        if (installButton) {
+            installButton.classList.remove('d-none');
+        }
+    }, 3000);  
+}
+
+window.AngorAPP = {
+    installPWA: function () {
+        if (window.PWADeferredPrompt) {
+            window.PWADeferredPrompt.prompt();
+            window.PWADeferredPrompt.userChoice
```

Comment:
Does this mean we tell the user there is a new version and they can decide to install or not?

Reply:
> pwa is refreshed and updated without the user noticing.
This is not so good I believe, we must discuss this actually.
Every time an update is available the user must be notified and they must trigger the update